tests/decap/test_decap.py

```diff
@@ -0,0 +1,242 @@
+        if "disabled_host_interfaces" in topology:
+            for intf in topology["disabled_host_interfaces"]:
+                downlink_ports.remove("[{}]".format(intf))
+    elif topo_type == "t1":
+        for k, v in topology["VMs"].items():
+            if "T0" in k:
+                downlink_ports.append("[{}]".format(" ".join([str(vlan) for vlan in v["vlans"]])))
+    return downlink_ports
+
+
+def gen_fib_info(ptfhost, testbed, cfg_facts):
+
+    topo_type = testbed["topo"]["type"]
+    topology = testbed["topo"]["properties"]["topology"]
+
+    # uplink ports
+    uplink_ports_str = " ".join(get_uplink_ports(topology, topo_type))
+
+    # downlink ports
+    downlink_ports_str = " ".join(get_downlink_ports(topology, topo_type))
+
+    fibs = []
+
+    podset_number = 5  # Limit the number of podsets to limit test execution time
+    tor_number = 16
+    tor_subnet_number = 2
+
+    # routes to uplink
+    routes_uplink_v4 = []
+    routes_uplink_v6 = []
+    if topo_type == "t0":
+        routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "")
+        routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "")
+    elif topo_type == "t1":
+        routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="spine")
+        routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="spine")
+
+    for prefix, _, _ in routes_uplink_v4:
+        fibs.append("{} {}".format(prefix, uplink_ports_str))
+    for prefix, _, _ in routes_uplink_v6:
+        fibs.append("{} {}".format(prefix, uplink_ports_str))
+
+    routes_downlink_v4 = []
+    routes_downlink_v6 = []
+    if topo_type == "t1":
+        routes_downlink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="tor")
+        routes_downlink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="tor")
+
+    for prefix, _, _ in routes_downlink_v4:
+        fibs.append("{} {}".format(prefix, downlink_ports_str))
+    for prefix, _, _ in routes_downlink_v6:
+        fibs.append("{} {}".format(prefix, downlink_ports_str))
+
+    ptfhost.copy(content="\n".join(fibs), dest="/root/fib_info.txt")
+
+
+def prepare_ptf(ptfhost, testbed, cfg_facts):
```

Comment:
Can we convert these into fixtures? They steps that are repeated in many test cases and it would better to invoke them once. This could be achieved by changing the fixture scope later on

Reply:
Good idea. I prefer to create another PR to generalize the steps for preparing PTF to a session scoped fixture and refactor the related scripts to use the fixture instead.